ccl/changefeedccl: TestChangefeedMonitoring failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/57274fc2cd56e5e57957db59acbbcf4f35c118bc

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestChangefeedMonitoring PKG=github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1229137&tab=buildLog

```
I190409 05:25:42.152416 39207 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 37376111854
I190409 05:25:43.152697 39207 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 38376392024
I190409 05:25:44.153092 39207 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 39376742503
I190409 05:25:44.627320 39327 server/status/runtime.go:500  [n1] runtime stats: 155 MiB RSS, 198 goroutines, 25 MiB/32 MiB/70 MiB GO alloc/idle/total, 16 MiB/47 MiB CGO alloc/total, 2501.5 CGO/sec, 17.3/7.3 %(u/s)time, 0.0 %gc (1x), 4.5 MiB/4.5 MiB (r/w)net
I190409 05:25:45.153394 39207 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 40377073947
I190409 05:25:46.153620 39207 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 41377327979
I190409 05:25:47.153867 39207 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 42377560374
I190409 05:25:48.154107 39207 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 43377809459
I190409 05:25:49.154329 39207 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 44378014291
I190409 05:25:50.155016 39207 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] stats.Refresher: maybeRefreshStats
1      [async] rangefeed: output loop
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
1      [async] changefeed-poller
I190409 05:25:50.155068 39207 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] stats.Refresher: maybeRefreshStats
1      [async] rangefeed: output loop
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
1      [async] changefeed-poller
I190409 05:25:50.155155 39207 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] rangefeed: output loop
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
1      [async] changefeed-poller
I190409 05:25:50.155213 39207 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber
1      [async] changefeed-poller
I190409 05:25:50.155382 39207 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] changefeed-poller
W190409 05:25:50.160407 39597 internal/client/txn.go:510  [n1,client=127.0.0.1:54024,user=root] failure aborting transaction: node unavailable; try another peer; abort caused by: context canceled
    --- FAIL: TestChangefeedMonitoring/sinkless (45.58s)
        changefeed_test.go:1010: condition failed to evaluate within 45s: waiting for the feed to be < 100000000 nanos behind got 44378014291
            goroutine 39207 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc002646510, 0x3918b80)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x39889a0, 0xc000bf4700, 0xc002011c00)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x103
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.TestChangefeedMonitoring.func1(0xc000bf4700, 0xc0010b3d40, 0x3920fe0, 0xc000dafa70)
            	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_test.go:1010 +0x90d
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.sinklessTest.func1(0xc000bf4700)
            	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:239 +0x5ad
            testing.tRunner(0xc000bf4700, 0xc0007c0040)
            	/usr/local/go/src/testing/testing.go:827 +0xbf
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x35c

```

SHA: https://github.com/cockroachdb/cockroach/commits/682c2f2f466bbf768545ca4687822206a63983ad

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestChangefeedMonitoring PKG=github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1231906&tab=buildLog

```
I190410 05:23:30.196968 39969 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 35369514190
I190410 05:23:31.197254 39969 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 36369804048
I190410 05:23:32.197515 39969 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 37370068033
I190410 05:23:33.197826 39969 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 38370366574
I190410 05:23:34.198083 39969 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 39370636200
I190410 05:23:34.687431 39728 server/status/runtime.go:500  [n1] runtime stats: 158 MiB RSS, 198 goroutines, 30 MiB/26 MiB/70 MiB GO alloc/idle/total, 16 MiB/53 MiB CGO alloc/total, 2511.1 CGO/sec, 14.5/7.9 %(u/s)time, 0.0 %gc (1x), 3.8 MiB/3.8 MiB (r/w)net
I190410 05:23:35.198365 39969 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 40370902434
I190410 05:23:36.198628 39969 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 41371183243
I190410 05:23:37.198914 39969 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 42371462745
I190410 05:23:38.199201 39969 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 43371746443
I190410 05:23:39.199490 39969 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 44372050143
I190410 05:23:40.200140 39969 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] stats.Refresher: maybeRefreshStats
1      [async] rangefeed: output loop
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
1      [async] changefeed-poller
I190410 05:23:40.200184 39969 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] stats.Refresher: maybeRefreshStats
1      [async] rangefeed: output loop
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
1      [async] changefeed-poller
I190410 05:23:40.200234 39969 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber
1      [async] changefeed-poller
I190410 05:23:40.200546 39969 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] changefeed-poller
W190410 05:23:40.206036 40398 internal/client/txn.go:510  [n1,client=127.0.0.1:50722,user=root] failure aborting transaction: node unavailable; try another peer; abort caused by: context canceled
    --- FAIL: TestChangefeedMonitoring/sinkless (45.55s)
        changefeed_test.go:1010: condition failed to evaluate within 45s: waiting for the feed to be < 100000000 nanos behind got 44372050143
            goroutine 39969 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc001abb260, 0x3919a40)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x3989840, 0xc000cf6000, 0xc000eb80a0)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x103
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.TestChangefeedMonitoring.func1(0xc000cf6000, 0xc00212c6c0, 0x3921ea0, 0xc002a12480)
            	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_test.go:1010 +0x90d
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.sinklessTest.func1(0xc000cf6000)
            	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:239 +0x5ad
            testing.tRunner(0xc000cf6000, 0xc0004a6bb0)
            	/usr/local/go/src/testing/testing.go:827 +0xbf
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x35c

```

SHA: https://github.com/cockroachdb/cockroach/commits/bf399d2677783dc1eea7f5ede6d4561f95c0ea10

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestChangefeedMonitoring PKG=github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1234715&tab=buildLog

```
I190411 05:32:08.058347 40165 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 37370282285
I190411 05:32:09.058573 40165 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 38370544519
I190411 05:32:10.058863 40165 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 39370833848
I190411 05:32:10.547955 40356 server/status/runtime.go:500  [n1] runtime stats: 164 MiB RSS, 198 goroutines, 19 MiB/100 MiB/137 MiB GO alloc/idle/total, 16 MiB/51 MiB CGO alloc/total, 2526.8 CGO/sec, 13.7/7.2 %(u/s)time, 0.0 %gc (2x), 2.6 MiB/2.6 MiB (r/w)net
I190411 05:32:11.059076 40165 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 40371048255
I190411 05:32:12.059299 40165 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 41371260343
I190411 05:32:13.059695 40165 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 42371619319
I190411 05:32:14.060038 40165 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 43371994721
I190411 05:32:15.060373 40165 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 44372333226
I190411 05:32:16.063101 40165 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] stats.Refresher: maybeRefreshStats
1      [async] rangefeed: output loop
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
1      [async] changefeed-poller
I190411 05:32:16.063529 40165 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] rangefeed: output loop
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
1      [async] changefeed-poller
I190411 05:32:16.063561 40165 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] rangefeed: output loop
1      [async] intent_resolver_ir_batcher
1      [async] closedts-rangefeed-subscriber
1      [async] changefeed-poller
I190411 05:32:16.063579 40165 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] rangefeed: output loop
1      [async] closedts-rangefeed-subscriber
1      [async] changefeed-poller
I190411 05:32:16.063762 40165 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber
1      [async] changefeed-poller
I190411 05:32:16.063952 40165 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] changefeed-poller
    --- FAIL: TestChangefeedMonitoring/sinkless (45.56s)
        changefeed_test.go:1010: condition failed to evaluate within 45s: waiting for the feed to be < 100000000 nanos behind got 44372333226
            goroutine 40165 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc00102fdd0, 0x391c500)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x398c2a0, 0xc00067ec00, 0xc000884940)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x103
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.TestChangefeedMonitoring.func1(0xc00067ec00, 0xc000a9c300, 0x3924960, 0xc000d0c990)
            	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_test.go:1010 +0x90d
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.sinklessTest.func1(0xc00067ec00)
            	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:239 +0x5ad
            testing.tRunner(0xc00067ec00, 0xc0005f14d0)
            	/usr/local/go/src/testing/testing.go:827 +0xbf
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x35c

```

SHA: https://github.com/cockroachdb/cockroach/commits/58a9bf5992326ca6549cd3933bac175ff28e9bfc

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestChangefeedMonitoring PKG=github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1237534&tab=buildLog

```
I190412 05:43:35.854329 41122 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 34369955680
I190412 05:43:36.854662 41122 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 35370282295
I190412 05:43:37.855080 41122 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 36370656214
I190412 05:43:38.855458 41122 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 37371071176
I190412 05:43:39.855796 41122 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 38371414322
I190412 05:43:40.856179 41122 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 39371792438
I190412 05:43:41.329319 41038 server/status/runtime.go:500  [n1] runtime stats: 170 MiB RSS, 198 goroutines, 20 MiB/98 MiB/137 MiB GO alloc/idle/total, 16 MiB/51 MiB CGO alloc/total, 2501.9 CGO/sec, 18.4/8.7 %(u/s)time, 0.0 %gc (2x), 1.5 MiB/1.5 MiB (r/w)net
I190412 05:43:41.856551 41122 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 40372152688
I190412 05:43:42.857164 41122 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 41372800155
I190412 05:43:43.857534 41122 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 42373136517
I190412 05:43:44.857831 41122 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 43373446631
I190412 05:43:45.858148 41122 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 44373765521
I190412 05:43:46.858944 41122 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] stats.Refresher: maybeRefreshStats
1      [async] rangefeed: output loop
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
1      [async] changefeed-poller
I190412 05:43:46.858985 41122 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] stats.Refresher: maybeRefreshStats
1      [async] rangefeed: output loop
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
1      [async] changefeed-poller
I190412 05:43:46.859073 41122 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber
1      [async] changefeed-poller
I190412 05:43:46.859284 41122 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] changefeed-poller
    --- FAIL: TestChangefeedMonitoring/sinkless (45.57s)
        changefeed_test.go:1010: condition failed to evaluate within 45s: waiting for the feed to be < 100000000 nanos behind got 44373765521
            goroutine 41122 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc00150c600, 0x395c860)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x39cc740, 0xc001332200, 0xc0018c68c0)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x103
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.TestChangefeedMonitoring.func1(0xc001332200, 0xc0010e4540, 0x3964cc0, 0xc0018e8fc0)
            	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_test.go:1010 +0x90d
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.sinklessTest.func1(0xc001332200)
            	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:239 +0x5ad
            testing.tRunner(0xc001332200, 0xc000097dd0)
            	/usr/local/go/src/testing/testing.go:827 +0xbf
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x35c

```

SHA: https://github.com/cockroachdb/cockroach/commits/c53084cbd4b2be90789652ffb74b3d0c02930f7c

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestChangefeedMonitoring PKG=github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1239665&tab=buildLog

```
I190413 05:17:55.288038 41527 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 37372756204
I190413 05:17:56.288357 41527 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 38373082988
I190413 05:17:57.288646 41527 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 39373372263
I190413 05:17:57.777996 41690 server/status/runtime.go:500  [n1] runtime stats: 146 MiB RSS, 198 goroutines, 20 MiB/35 MiB/70 MiB GO alloc/idle/total, 15 MiB/51 MiB CGO alloc/total, 2511.9 CGO/sec, 15.9/6.9 %(u/s)time, 0.0 %gc (2x), 2.2 MiB/2.2 MiB (r/w)net
I190413 05:17:58.288950 41527 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 40373664393
I190413 05:17:59.289317 41527 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 41374035291
I190413 05:18:00.289659 41527 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 42374399995
I190413 05:18:01.290009 41527 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 43374739926
I190413 05:18:02.290320 41527 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 44375035173
I190413 05:18:03.290911 41527 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] stats.Refresher: maybeRefreshStats
1      [async] rangefeed: output loop
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
1      [async] changefeed-poller
I190413 05:18:03.290956 41527 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] stats.Refresher: maybeRefreshStats
1      [async] rangefeed: output loop
1      [async] intent_resolver_ir_batcher
1      [async] closedts-rangefeed-subscriber
1      [async] changefeed-poller
I190413 05:18:03.290981 41527 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] stats.Refresher: maybeRefreshStats
1      [async] rangefeed: output loop
1      [async] closedts-rangefeed-subscriber
1      [async] changefeed-poller
I190413 05:18:03.291092 41527 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] stats.Refresher: maybeRefreshStats
1      [async] closedts-rangefeed-subscriber
1      [async] changefeed-poller
I190413 05:18:03.291130 41527 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber
1      [async] changefeed-poller
I190413 05:18:03.291179 41527 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] changefeed-poller
    --- FAIL: TestChangefeedMonitoring/sinkless (45.54s)
        changefeed_test.go:1010: condition failed to evaluate within 45s: waiting for the feed to be < 100000000 nanos behind got 44375035173
            goroutine 41527 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc0009063f0, 0x395d2e0)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x39cd1c0, 0xc000a3e100, 0xc0007a9380)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x103
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.TestChangefeedMonitoring.func1(0xc000a3e100, 0xc001410600, 0x3965740, 0xc00211fcb0)
            	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_test.go:1010 +0x90d
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.sinklessTest.func1(0xc000a3e100)
            	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:239 +0x5ad
            testing.tRunner(0xc000a3e100, 0xc0020dc0c0)
            	/usr/local/go/src/testing/testing.go:827 +0xbf
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x35c

```

SHA: https://github.com/cockroachdb/cockroach/commits/ee451b4f273e93018969bc0edd34e7105625adaf

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestChangefeedMonitoring PKG=github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1240757&tab=buildLog

```
I190414 05:15:18.942096 40247 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 35367952195
I190414 05:15:19.942504 40247 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 36368360125
I190414 05:15:20.942805 40247 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 37368667176
I190414 05:15:21.943053 40247 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 38368926458
I190414 05:15:22.943299 40247 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 39369161622
I190414 05:15:23.421120 40401 server/status/runtime.go:500  [n1] runtime stats: 176 MiB RSS, 198 goroutines, 19 MiB/99 MiB/137 MiB GO alloc/idle/total, 16 MiB/48 MiB CGO alloc/total, 2502.3 CGO/sec, 19.5/7.8 %(u/s)time, 0.0 %gc (2x), 5.3 MiB/5.3 MiB (r/w)net
I190414 05:15:23.917771 40349 storage/queue.go:435  [n1,s1,r2/1:/System/NodeLiveness{-Max}] rate limited in maybeadd-merge: throttled on async limiting semaphore
I190414 05:15:23.943570 40247 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 40369420326
I190414 05:15:24.943829 40247 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 41369686617
I190414 05:15:25.944080 40247 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 42369943314
I190414 05:15:26.944334 40247 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 43370200695
I190414 05:15:27.944586 40247 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 44370447796
I190414 05:15:28.945236 40247 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] stats.Refresher: maybeRefreshStats
1      [async] rangefeed: output loop
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
1      [async] changefeed-poller
I190414 05:15:28.945275 40247 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] stats.Refresher: maybeRefreshStats
1      [async] rangefeed: output loop
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
1      [async] changefeed-poller
I190414 05:15:28.945408 40247 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber
1      [async] changefeed-poller
I190414 05:15:28.948164 40247 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] changefeed-poller
    --- FAIL: TestChangefeedMonitoring/sinkless (45.56s)
        changefeed_test.go:1010: condition failed to evaluate within 45s: waiting for the feed to be < 100000000 nanos behind got 44370447796
            goroutine 40247 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc00064ae10, 0x395df20)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x39cde00, 0xc000a3a200, 0xc001442fc0)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x103
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.TestChangefeedMonitoring.func1(0xc000a3a200, 0xc0021fe0c0, 0x3966380, 0xc00300f680)
            	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_test.go:1010 +0x90d
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.sinklessTest.func1(0xc000a3a200)
            	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:239 +0x5ad
            testing.tRunner(0xc000a3a200, 0xc0004a8710)
            	/usr/local/go/src/testing/testing.go:827 +0xbf
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x35c

```

SHA: https://github.com/cockroachdb/cockroach/commits/d3f704f839ccaef7f10c3af48c78a26d390ae1dc

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestChangefeedMonitoring PKG=github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1241469&tab=buildLog

```
I190415 05:15:04.676679 40281 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 35371831423
I190415 05:15:05.676932 40281 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 36372066359
I190415 05:15:06.677159 40281 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 37372313333
I190415 05:15:07.677382 40281 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 38372529561
I190415 05:15:08.677598 40281 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 39372744516
I190415 05:15:09.164586 40510 server/status/runtime.go:500  [n1] runtime stats: 148 MiB RSS, 198 goroutines, 20 MiB/36 MiB/70 MiB GO alloc/idle/total, 16 MiB/52 MiB CGO alloc/total, 2524.2 CGO/sec, 14.0/6.6 %(u/s)time, 0.0 %gc (2x), 5.4 MiB/5.4 MiB (r/w)net
I190415 05:15:09.677856 40281 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 40373012859
I190415 05:15:10.678062 40281 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 41373205324
I190415 05:15:11.678328 40281 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 42373482294
I190415 05:15:12.678544 40281 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 43373691793
I190415 05:15:13.678757 40281 ccl/changefeedccl/changefeed_test.go:1010  SucceedsSoon: waiting for the feed to be < 100000000 nanos behind got 44373909606
I190415 05:15:14.679317 40281 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] stats.Refresher: maybeRefreshStats
1      [async] rangefeed: output loop
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
1      [async] changefeed-poller
I190415 05:15:14.679364 40281 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] stats.Refresher: maybeRefreshStats
1      [async] rangefeed: output loop
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
1      [async] changefeed-poller
I190415 05:15:14.679443 40281 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber
1      [async] changefeed-poller
I190415 05:15:14.679653 40281 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] changefeed-poller
W190415 05:15:14.687592 40890 internal/client/txn.go:510  [n1,client=127.0.0.1:60998,user=root] failure aborting transaction: node unavailable; try another peer; abort caused by: context canceled
    --- FAIL: TestChangefeedMonitoring/sinkless (45.55s)
        changefeed_test.go:1010: condition failed to evaluate within 45s: waiting for the feed to be < 100000000 nanos behind got 44373909606
            goroutine 40281 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc0012ae690, 0x395df20)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x39cde00, 0xc00112c000, 0xc001ae6ea0)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x103
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.TestChangefeedMonitoring.func1(0xc00112c000, 0xc000aa5980, 0x3966380, 0xc000ea1050)
            	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_test.go:1010 +0x90d
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.sinklessTest.func1(0xc00112c000)
            	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:239 +0x5ad
            testing.tRunner(0xc00112c000, 0xc002252000)
            	/usr/local/go/src/testing/testing.go:827 +0xbf
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x35c

```

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -1005,7 +1005,7 @@ func TestChangefeedMonitoring(t *testing.T) {
 
 		// Unblocking the emit should bring the max_behind_nanos back down and
 		// should update the min_high_water.
-		beforeEmitRowCh <- struct{}{}
+		close(beforeEmitRowCh)
 		_, _ = foo.Next()
 		testutils.SucceedsSoon(t, func() error {
 			waitForBehindNanos := expectedLatency.Nanoseconds()
@@ -1023,26 +1023,25 @@ func TestChangefeedMonitoring(t *testing.T) {
 		})
 
 		// Check that two changefeeds add correctly.
-		beforeEmitRowCh <- struct{}{}
-		beforeEmitRowCh <- struct{}{}
 		// Set cluster settings back so we don't interfere with schema changes.
 		sqlDB.Exec(t, `SET CLUSTER SETTING kv.closed_timestamp.target_duration = '1s'`)
 		fooCopy := feed(t, f, `CREATE CHANGEFEED FOR foo`)
 		_, _ = fooCopy.Next()
 		_, _ = fooCopy.Next()
 		testutils.SucceedsSoon(t, func() error {
-			if c := s.MustGetSQLCounter(`changefeed.emitted_messages`); c != 4 {
-				return errors.Errorf(`expected 4 got %d`, c)
+			// We can't assert exactly 4 or 88 in case we get (allowed) duplicates
+			// from RangeFeed.
+			if c := s.MustGetSQLCounter(`changefeed.emitted_messages`); c < 4 {
+				return errors.Errorf(`expected >= 4 got %d`, c)
 			}
-			if c := s.MustGetSQLCounter(`changefeed.emitted_bytes`); c != 88 {
-				return errors.Errorf(`expected 88 got %d`, c)
+			if c := s.MustGetSQLCounter(`changefeed.emitted_bytes`); c < 88 {
+				return errors.Errorf(`expected >= 88 got %d`, c)
 			}
 			return nil
 		})
 
 		// Cancel all the changefeeds and check that max_behind_nanos returns to 0
 		// and min_high_water returns to the no high-water sentinel.
-		close(beforeEmitRowCh)
 		require.NoError(t, foo.Close())
 		require.NoError(t, fooCopy.Close())
 		testutils.SucceedsSoon(t, func() error {
@@ -1056,10 +1055,7 @@ func TestChangefeedMonitoring(t *testing.T) {
 		})
 	}
 
-	t.Run(`sinkless`, func(t *testing.T) {
-		t.Skip("https://github.com/cockroachdb/cockroach/issues/36643")
-		sinklessTest(testFn)
-	})
+	t.Run(`sinkless`, sinklessTest(testFn))
 	t.Run(`enterprise`, enterpriseTest(testFn))
 	t.Run(`poller`, pollerTest(sinklessTest, testFn))
 }

